src/core/server/http/http_server.ts

```diff
@@ -129,13 +129,20 @@ export class HttpServer {
       for (const route of router.getRoutes()) {
         this.log.debug(`registering route handler for [${route.path}]`);
         const { authRequired = true, tags } = route.options;
+        // Hapi does not allow payload validation to be specified for 'head' or 'get' requests
+        const validate = ['head', 'get'].includes(route.method) ? undefined : { payload: true };
         this.server.route({
           handler: route.handler,
           method: route.method,
           path: route.path,
           options: {
             auth: authRequired ? undefined : false,
             tags: tags ? Array.from(tags) : undefined,
+            // TODO: This 'validate' section can be removed once the legacy platform is completely removed.
+            // We are telling Hapi that NP routes can accept any payload, so that it can bypass the default
+            // validation applied in ./http_tools#getServerOptions
+            // (All NP routes are already required to specify their own validation in order to access the payload)
```

Comment:
not sure that we can remove it. NP plugins might disable validation https://github.com/restrry/kibana/blob/841abd1162f1eb6aaebf27d004c92e8344636d91/src/core/server/http/router/route.ts#L78-L80

Reply:
Oh thanks for the heads-up! Are we sure we want to allow this behavior? Do we have concrete use cases for disabling the `config-schema` validation at this time?

Reply:
There are some cases when we don't know the shape of an object upfront.
https://github.com/restrry/kibana/blob/841abd1162f1eb6aaebf27d004c92e8344636d91/x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js#L204
https://github.com/elastic/kibana/pull/48413/files#diff-fbd7aa2d4390350fcd1ffd047c33af25

Reply:
> There are some cases when we don't know the shape of an object upfront.

That's a fair point -- we do know that payloads will at least be objects or arrays though, right? Could we at least enforce something like `schema.object({}, { allowUnknowns: true })` at route handlers?

We are proposing a followup PR to this one which will incorporate these default validations into the `schema.object` call itself, so that all NP routes will have these protections in place, unless they explicitly opt-out of them via `schema.unsafeObject({})` or similar.

Reply:
> Could we at least enforce something like schema.object({}, { allowUnknowns: true }) at route handlers?

We already enforce. Don't we? Plugins cannot get access to a body without declaring validation
https://github.com/elastic/kibana/blob/master/src/core/server/http/router/request.ts#L106-L109
Although they can lax the validation
```js
schema.any({ validate: obj => obj })
schema.object({}, { validate: obj => obj })
schema.object({}, { allowUnknowns: true })
```

Reply:
Ah sorry I misunderstood. I thought NP provided a way to prevent any validation (not even `schema.any()`, and still get access to the underlying payload. I think we're OK as-is then. Our followup PR should protect these routes as well.